examples/hhl.py

```diff
@@ -0,0 +1,414 @@
+
+            for q in qubits[-2::-1]:
+                # Place X gates
+                if xor % 2 == 1:
+                    yield cirq.X(q)
+                xor >>= 1
+
+                # Build controlled ancilla rotation
+                kGate = cirq.ControlledGate(kGate)
+
+            yield kGate(*qubits)
+
+    def __ancilla_rotation(self, k):
+        if k == 0:
+            k = self.N
+        theta = 2*math.asin(self.C * self.N * self.t / (2*math.pi * k))
+        return cirq.Ry(theta)
+
+
+def hhl_circuit(A, C, t, register_size, *input_prep_gates):
+    """
+    Constructs the HHL circuit.
+
+    A is the input Hermitian matrix.
+    C and t are tunable parameters for the algorithm.
+    register_size is the size of the eigenvalue register.
+    input_prep_gates is a list of gates to be applied to |0> to generate the
+      desired input state |b>.
+    """
+
+    ancilla = cirq.GridQubit(0, 0)
+    # to store eigenvalues of the matrix
+    register = [cirq.GridQubit(i+1, 0) for i in range(register_size)]
+    # to store input and output vectors
+    memory = cirq.GridQubit(register_size+1, 0)
+
+    c = cirq.Circuit()
+    hs = HamiltonianSimulation(A, t)
+    pe = PhaseEstimation(register_size+1, hs)
+    c.append([gate(memory) for gate in input_prep_gates])
+    c.append([
+        pe(*register, memory),
+        EigenRotation(register_size+1, C, t)(*register, ancilla),
+        (pe**-1)(*register, memory),
+        cirq.measure(ancilla)
+    ])
+
+    # Pauli observable display
+    c.append([
+        cirq.pauli_string_expectation(
+            cirq.PauliString({ancilla: cirq.Z}),
+            key='a'
+        ),
+        cirq.pauli_string_expectation(
+            cirq.PauliString({memory: cirq.X}),
+            key='x'
+        ),
+        cirq.pauli_string_expectation(
+            cirq.PauliString({memory: cirq.Y}),
+            key='y'
+        ),
+        cirq.pauli_string_expectation(
+            cirq.PauliString({memory: cirq.Z}),
+            key='z'
+        ),
+    ])
+
+    return c
+
+
+def simulate(circuit):
+    simulator = cirq.Simulator()
+
+    # TODO optimize using amplitude amplification algorithm
+    ancilla_expectation = 0.0
+    while ancilla_expectation != -1.0:
+        result = simulator.compute_displays(circuit)
+        ancilla_expectation = round(result.display_values['a'], 3)
+
+    # Compute displays
+    print('X = {}'.format(result.display_values['x']))
+    print('Y = {}'.format(result.display_values['y']))
+    print('Z = {}'.format(result.display_values['z']))
+
+
+def main():
+    """
+    Simulates HHL with input given in each test case, and outputs Pauli
+    observables of the resulting qubit state |x>.
+    Expected observables are calculated from the expected solution |x>.
+    """
+
+    testcases = [
+        {
+            # Eigendecomposition: [(2, |+>), (1, |->)
+            # |b> = |0>
+            # |x> = (0.948683, -0.316228)
+            'A': np.array([[1.5, 0.5], [0.5, 1.5]]),
```

Comment:
I would move these checks into the tests. For the example you should just take one matrix and show that it prints out the expected stuff.